vue/components/ui/molecules/table/table.vue

```diff
@@ -308,9 +377,73 @@ export const Table = {
             };
             if (this.variant) base[this.variant] = true;
             return base;
+        },
+        isAllChecked() {
+            return (
+                Object.values(this.checkedItemsData).length > 0 &&
+                Object.values(this.checkedItemsData).length === this.itemsData.length &&
+                !Object.values(this.checkedItemsData).includes(false)
+            );
+        },
+        isAllUnchecked() {
+            return !Object.values(this.checkedItemsData).includes(true);
         }
     },
     methods: {
+        itemsWithIndex() {
+            return this.items.map((item, index) => ({ _originalIndex: index, ...item }));
+        },
+        isRowSelected(originalIndex) {
+            return this.allowSelectedHighlight && originalIndex === this.selectedOriginalIndex;
+        },
+        itemsChangeHandler(items, itemsNrDiff) {
+            if (itemsNrDiff >= 0) {
+                let item = null;
+                for (let i = items.length - itemsNrDiff; i < items.length; i++) {
+                    item = items[i];

```

Comment:
maybe remove the "let item = null" from above and use "const item = item[i]" inside the loop